src/Symfony/Component/Config/Resource/GlobResource.php

```diff
@@ -95,11 +99,27 @@ public function getIterator()
 
         if (0 !== strpos($this->prefix, 'phar://') && false === strpos($this->pattern, '/**/') && (\defined('GLOB_BRACE') || false === strpos($this->pattern, '{'))) {
             foreach (glob($this->prefix.$this->pattern, \defined('GLOB_BRACE') ? GLOB_BRACE : 0) as $path) {
+                if (is_dir($path)) {
+                    if ($this->forExclusion && \in_array($path, glob($this->prefix.$this->pattern, GLOB_BRACE | GLOB_ONLYDIR))) {
```

Comment:
GLOB_BRACE should be use conditionally as done just above
in_array should also be provided with its 3rd argument (true = strict)

Reply:
What's the deal with `GLOB_BRACE`? Is it because `The GLOB_BRACE flag is not available on some non GNU systems, like Solaris. `?

Reply:
Correct.

Reply:
But then, shouldn't we throw an error message if the `{` is in the `pattern` and `GLOB_BRACE` is not defined? 
Otherwise it won't behave as expected and user won't be aware of it.

Reply:
That's already dealt with, see L100. But this also means we might have forgotten the czse below this "if" that falls back to using Finder. Does need to know about excluded paths also?

Reply:
Initially it wasn't aware of excluded prefixes, but then it made some tests fail eg:
`src/Symfony/Component/DependencyInjection/Tests/Loader/YamlFileLoaderTest.php` for `testPrototype`.
In that test `'../Prototype/{OtherDir,BadClasses}'` is excluded, so on first foreach loop as `src/Symfony/Component/DependencyInjection/Tests/Fixtures/Prototype` is not in excluded prefixes, it gets to line 113, so anything under that folder is not longer parsed by the initial `glob` foreach, but by the recursive iterator.
It's taken me quite a lot of time to figure all this out, in fact this class is getting quite complex, so maybe it's worth a try to refactor it into something easit to follow?